File: app/src/main/java/com/example/android/moviesremake/retrofit/ApiClient.java
package com.example.android.moviesremake.retrofit;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.v7.preference.PreferenceManager;

import com.example.android.moviesremake.R;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by PepovPC on 1/16/2018.
 */

public class ApiClient {


    public static final String BASE_URL = "http://api.themoviedb.org/3/";
    private static Retrofit retrofit = null;


    public static Retrofit getClient() {
        if (retrofit==null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }


    public static String getSearchQuery(Context context) {
        SharedPreferences prefs = PreferenceManager
                .getDefaultSharedPreferences(context);
        String keyForUnits = context.getString(R.string.pref_units_key);
        String defaultUnits = context.getString(R.string.pref_units_metric);
        String preferredUnits = prefs.getString(keyForUnits, defaultUnits);
        String metric = context.getString(R.string.pref_units_metric);
        String imperial = context.getString(R.string.pref_units_imperial);
        String userPrefersMetric;
        if (metric.equals(preferredUnits)) {
            userPrefersMetric = defaultUnits;
        } else {
            userPrefersMetric = imperial;
        }
        return userPrefersMetric;
    }


}

File: app/src/main/java/com/example/android/moviesremake/retrofit/ApiClient.kt
package com.example.android.moviesremake.retrofit

import android.content.Context
import android.support.v7.preference.PreferenceManager
import com.example.android.moviesremake.R
import com.jakewharton.retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by PepovPC on 1/16/2018.
 */

object ApiClient {


    val BASE_URL = "http://api.themoviedb.org/3/"
    private var retrofit: Retrofit? = null


    val client: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofit
        }


    fun getSearchQuery(context: Context): String {
        val prefs = PreferenceManager
                .getDefaultSharedPreferences(context)
        val keyForUnits = context.getString(R.string.pref_units_key)
        val defaultUnits = context.getString(R.string.pref_units_metric)
        val preferredUnits = prefs.getString(keyForUnits, defaultUnits)
        val metric = context.getString(R.string.pref_units_metric)
        val imperial = context.getString(R.string.pref_units_imperial)
        val userPrefersMetric: String
        if (metric == preferredUnits) {
            userPrefersMetric = defaultUnits
        } else {
            userPrefersMetric = imperial
        }
        return userPrefersMetric
    }


}
